comp/logs/agent/config/config.go

```diff
@@ -160,6 +165,97 @@ func IsExpectedTagsSet(coreConfig pkgconfigmodel.Reader) bool {
 	return ExpectedTagsDuration(coreConfig) > 0
 }
 
+// applyFingerprintDefaults applies the default values to a fingerprint config
+func applyFingerprintDefaults(config *FingerprintConfig) {
+	config.FingerprintStrategy = pkgconfigsetup.DefaultFingerprintStrategy
+	config.Count = pkgconfigsetup.DefaultFingerprintingMaxLines
+	config.CountToSkip = pkgconfigsetup.DefaultLinesOrBytesToSkip
+	config.MaxBytes = pkgconfigsetup.DefaultFingerprintingMaxBytes
+}
+
+// GlobalFingerprintConfig returns the global fingerprint configuration to apply to all logs.
+func GlobalFingerprintConfig(coreConfig pkgconfigmodel.Reader) (*FingerprintConfig, error) {
+	var err error
+
+	// Get the complete fingerprint config map without applying defaults immediately
+	raw := coreConfig.Get("logs_config.fingerprint_config")
+	if raw == nil {
+		return nil, nil
+	}
+
+	// Create a config without defaults first
+	config := &FingerprintConfig{}
+
+	if s, ok := raw.(string); ok && s != "" {
+		log.Debugf("GlobalFingerprintConfig: unmarshaling from string: %s", s)
+		err = json.Unmarshal([]byte(s), &config)
+	} else {
+		log.Debugf("GlobalFingerprintConfig: using structure.UnmarshalKey")
+		// Fallback to structure.UnmarshalKey
+		err = structure.UnmarshalKey(coreConfig, "logs_config.fingerprint_config", &config, structure.ConvertEmptyStringToNil)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("GlobalFingerprintConfig: after unmarshaling - FingerprintStrategy: %s, Count: %d, CountToSkip: %d, MaxBytes: %d",
+		config.FingerprintStrategy, config.Count, config.CountToSkip, config.MaxBytes)
+
+	// Validate the config
+	err = ValidateFingerprintConfig(config)
+	if err != nil {
+		if errors.Is(err, ErrEmptyFingerprintConfig) {
+			// If config is empty, apply all defaults
+			applyFingerprintDefaults(config)
+			return config, nil
+		}
+		return nil, err
+	}
+
+	return config, nil
+}
+
+// ValidateFingerprintConfig validates the fingerprint config and returns an error if the config is invalid
+func ValidateFingerprintConfig(config *FingerprintConfig) error {
+	if config == nil {
+		return ErrEmptyFingerprintConfig
+	}
+
```

Comment:
If we were to not take a pointer we would not have to worry about `nil` here. Seems like nowhere in the code we pass a nil pointer to this func.